x/txfees/keeper/sybil.go

```diff
@@ -0,0 +1,30 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type Sybil struct {
+	GasPrice sdk.Dec
+	FeesPaid sdk.Coin
+}
+
+func NewSybil(gasPrice sdk.Dec, feesPaid sdk.Coin) Sybil {
+	return Sybil{
+		GasPrice: gasPrice,
+		FeesPaid: feesPaid,
+	}
+}
+
+func (s Sybil) AddToFeesPaid(feesPaid sdk.Coin) (Sybil, error) {
+	// Check same denom
+	if feesPaid.Denom != s.FeesPaid.Denom {
+		return Sybil{}, fmt.Errorf("Cannot add %s denom to sybil's %s fees paid denom", feesPaid.Denom, s.FeesPaid.Denom)
+	}
+	// Add tokens
+	fp := s.FeesPaid.Add(feesPaid)
+	// Return new sybil with tokens added together & previous gas price
+	return Sybil{GasPrice: s.GasPrice, FeesPaid: fp}, nil
```

Comment:
What is the purpose for returning a new `Sybil`?

It seems to me that we can make `(s *Sybil)` (pointer receiver) and modify the existing instance without having to create new